object/error.go

```diff
@@ -10,10 +12,13 @@ func (s *SplitInfoError) Error() string {
 	return splitInfoErrorMsg
 }
 
+// SplitInfo returns object SplitInfo.
 func (s *SplitInfoError) SplitInfo() *SplitInfo {
 	return s.si
 }
 
+// NewSplitInfoError creates new SplitInfoError
+// with provided SplitInfo.
 func NewSplitInfoError(v *SplitInfo) *SplitInfoError {
```

Comment:
Lets provide setter and remove constructor.

Reply:
delete even constructor for an error? i do not agree with that

most use cases are just returning `SplitInfoError` if we faced `SplitInfo`, so i do not see any preferences for using 3 lines of code (`var`, setter, `return`) instead of one (just returning with errConstructor): can not imaging that we could reuse one var via multiple setter calls

Reply:
Then I suggest changing the constructor return to `error`.